### ISSUE TITLE
Baseline browser release channels?

### ISSUE THREAD

**Issue by romainmenke** [web-platform-dx/web-features#534]
The current efforts and outcomes around Baseline focus on the Baseline definition itself and on documentation and mapping of features.
But most developers do not lookup the support status of each feature they use.

While discussing this we were wondering if browsers like Chrome and Firefox which aren't tied to an OS could have a "Baseline release channel". This would be releases which are always 30 months behind.

Firefox old releases : https://ftp.mozilla.org/pub/firefox/releases/
Chrome doesn't have a list but it is possible : https://www.browserstack.com/guide/downgrade-to-older-versions-of-chrome

Developers could then more easily install such a Baseline release and test their projects.

Working in such a browser version enables a more intuitive trial and error workflow for developers.

_I thought I'd raise this issue here first to get a reading on why this is a bad idea, or maybe a great idea._

**Comment by foolip** [comment]
Something that might work for JS and APIs is to remove all of the APIs that aren't Baseline before any other code runs. This could be done in testing setups. The setup code would be something like this:

```js
const globalAPIs = ['showDirectoryPicker', '...'];
for (const prop of globalAPIs) {
  Object.defineProperty(globalThis, prop, {value: undefined});
}
const navigatorAPIs = ['xr', '...'];
for (const prop of navigatorAPIs) {
  Object.defineProperty(navigator, prop, {value: undefined});
}
```

I haven't tried this in practice and I think it's not easy to fully erase some APIs, but it might be useful.

**Comment by romainmenke** [comment]
That would require "reverse-polyfills".
Larger features (a new class) can be easily deleted but everything else would require wrapping implementations to erase new arguments or new kinds of values.

I think this easily becomes more trouble than it's worth :)

**Comment by captainbrosset** [comment]
Agreed that "most developers do not lookup the support status of each feature they use" needs to be something the WebDX group takes into account this year when working on Baseline. I'm not sold on the idea that browsers would maintain a channel that matches the current Baseline. But I'm very much for more and simpler tooling that's based on the data. Static analysis of the code, API explorer/catalog, etc.

**Comment by romainmenke** [comment]
> But I'm very much for more and simpler tooling that's based on the data. Static analysis of the code, API explorer/catalog, etc.

These tools exist and it would be awesome if this working group could collaborate with the maintainers of those tools. But these are very limited and can only give basic feedback.

The absolute best way to check if something works in a given browser version is by actually testing it in that version.

Currently this is a service that 3rd parties like Browserstack provide through VM's that contain older operating systems, browser versions, ... But this is also cumbersome to work with because the browser isn't running locally.

**Comment by romainmenke** [comment]
Maybe a non-technical approach is sufficient?

If someone want to deliver a project that "satisfies Baseline" they would need to check in specific browser versions when they first deliver the project and again with major changes.

How do you find what those browser versions are at any given point in time?

If this information is readily available somewhere developers could then use the method of their choosing to get access to those versions. Might be a manual setup, might be service like Browserstack.

**Comment by captainbrosset** [comment]
This new page on the site might be helpful: https://web-platform-dx.github.io/web-features/supported-browsers/
Closing based on this, but feel free to comment if you believe this should remain open.

**Comment by rviscomi** [comment]
https://github.com/web-platform-dx/baseline-browser-mapping powers the Supported Browsers page mentioned above and is also available as an npm package if you need to programmatically get the list of browsers for a given Baseline target (newly/widely/year/date)

cc @tonypconway
